### PR TITLE
feat(api): add Eventuras.Libs.DocComposer for Fluid template rendering

### PR DESCRIPTION
**Pull request by losolio**
## Summary
- New standalone library `Eventuras.Libs.DocComposer` for rendering localized Liquid templates with Fluid.
- Core `IDocumentComposer` returns rendered HTML, suitable for emails, certificates, invoices, and other document outputs.
- Templates are loaded per locale via `IFileProvider` with convention `{name}.{locale}.liquid` and fall back to the configured default locale when the requested one is missing. Locale lookup is case-insensitive.
- Thin `IEmailComposer` wrapper layers email-specific concerns on top: subject extracted from rendered `<title>`, plain-text body derived via an AngleSharp-based extractor (handles links with hrefs, images with alt, lists, scripts/styles stripped, whitespace collapsed).
- Uses Fluid's default `MemberAccessStrategy` with per-model-type registration on first use, so templates can only read properties on the models they're rendered with.
- DI extension `AddFluidDocComposer(...)` registers both composers with options validation that fails fast on host startup. Test project covers rendering, locale fallback, locale casing, configuration validation, email subject/text extraction, and plain-text extractor edge cases.

## Scope of this PR
Infrastructure only — neither composer is wired into a sending flow yet.

## Test plan
- [x] `dotnet test` for the new test project — 16/16 passing locally
- [x] CI: full solution build with the new projects added to `Eventuras.slnx`
- [x] CI: Docker build with `Eventuras.Libs.DocComposer` csproj copied in restore layer

### CODE REVIEW

File: apps/api/src/Eventuras.Libs.EmailComposer/FluidEmailComposer.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            throw new InvalidOperationException(
+                $"{nameof(EmailComposerOptions.TemplateFileProvider)} must be configured.");
+        }
+
+        _parser = new FluidParser();
+        _templateOptions = new TemplateOptions
+        {
+            MemberAccessStrategy = new UnsafeMemberAccessStrategy()
+        };
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.EmailComposer/FluidEmailComposer.cs]: Fixed in 08bd048: switched to the default `MemberAccessStrategy` and now register the model type per `ComposeAsync<TModel>` call (with a `ConcurrentDictionary` to avoid re-registration). Templates can no longer reflect over arbitrary members.

File: apps/api/src/Eventuras.Libs.DocComposer/FluidDocumentComposer.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    public FluidEmailComposer(IOptions<EmailComposerOptions> options)
+    {
+        _options = options.Value;
+        if (_options.TemplateFileProvider is null)
+        {
+            throw new InvalidOperationException(
+                $"{nameof(EmailComposerOptions.TemplateFileProvider)} must be configured.");
+        }
+
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.DocComposer/FluidDocumentComposer.cs:31]: Fixed in 08bd048: `DefaultLocale` is now validated in the constructor and throws `InvalidOperationException` if null/whitespace. Added a regression test.

File: apps/api/src/Eventuras.Libs.EmailComposer/PlainTextExtractor.cs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        {
+            return string.Empty;
+        }
+
+        var configuration = Configuration.Default;
+        var context = BrowsingContext.New(configuration);
+        var document = context.OpenAsync(req => req.Content(html)).GetAwaiter().GetResult();
+
+        INode root = document.Body ?? (INode)document.DocumentElement;
+        return ExtractFromNode(root);
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.EmailComposer/PlainTextExtractor.cs]: Fixed in 08bd048: `Extract` is now `ExtractAsync` and awaits `OpenAsync` properly. Tests updated to await it.

File: apps/api/src/Eventuras.Libs.DocComposer/PlainTextExtractor.cs [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        switch (node.NodeType)
+        {
+            case NodeType.Text:
+                sb.Append(node.TextContent);
+                return;
+
+            case NodeType.Element:
+                WriteElement((IElement)node, sb);
+                return;
+        }
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.DocComposer/PlainTextExtractor.cs:67]: Fixed in 08bd048: `WriteNode` now handles `Document` and `DocumentFragment` by iterating their child nodes.

File: apps/api/src/Eventuras.Libs.EmailComposer/PlainTextExtractor.cs [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        "h1", "h2", "h3", "h4", "h5", "h6",
+        "ul", "ol", "table", "tr", "blockquote", "pre"
+    };
+
+    public static string Extract(string html)
+    {
+        if (string.IsNullOrWhiteSpace(html))
+        {
+            return string.Empty;
+        }
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.EmailComposer/PlainTextExtractor.cs]: Fixed in 08bd048: signature is now `ExtractAsync(string? html, ...)`, matching the actual supported inputs.

File: apps/api/src/Eventuras.Libs.EmailComposer/FluidEmailComposer.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        var primary = fileProvider.GetFileInfo(primaryPath);
+        if (primary.Exists)
+        {
+            return await ReadAllTextAsync(primary, cancellationToken);
+        }
+
+        if (!string.Equals(locale, _options.DefaultLocale, StringComparison.OrdinalIgnoreCase))
+        {
+            var fallbackPath = BuildPath(templateName, _options.DefaultLocale);
+            var fallback = fileProvider.GetFileInfo(fallbackPath);
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.EmailComposer/FluidEmailComposer.cs]: Fixed in 08bd048: locales are normalized via `ToLowerInvariant()` before building the template path (and before comparing against `DefaultLocale`). Added a regression test with locale `"EN"` that now resolves `welcome.en.liquid`.

File: apps/api/src/Eventuras.Libs.DocComposer/PlainTextExtractor.cs [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+                {
+                    builder.Append('\n');
+                }
+            }
+            else
+            {
+                blankLines = 0;
+                builder.Append(line).Append('\n');
+            }
+        }
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.DocComposer/PlainTextExtractor.cs:175]: Not taking this one — the existing `foreach (var raw in lines) { var line = CollapseInlineWhitespace(raw).TrimEnd(); ... }` reads fine and a `.Select(...)` projection inside the loop header doesn't make it clearer.

File: apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace Eventuras.Libs.EmailComposer.Tests;
+
+public class FluidEmailComposerTest
+{
+    public sealed record WelcomeModel(string Name, string EventTitle, string Url);
+
+    private static FluidEmailComposer CreateComposer(string defaultLocale = "en")
+    {
+        var templatesDir = Path.Combine(AppContext.BaseDirectory, "Templates");
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs:20]: Not taking this one — `"Templates"` is a hardcoded literal, not user input, so `Path.Combine` cannot silently drop the base directory here. False positive.

File: apps/api/src/Eventuras.Libs.DocComposer/FluidDocumentComposer.cs [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(templateName);
+        ArgumentException.ThrowIfNullOrWhiteSpace(locale);
+
+        var templateSource = await LoadTemplateSourceAsync(templateName, locale, cancellationToken);
+        var template = _parser.Parse(templateSource);
+
+        RegisterModelType(typeof(TModel));
+        var context = new TemplateContext(model, _templateOptions);
+        var html = await template.RenderAsync(context);
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.DocComposer/FluidDocumentComposer.cs:57]: Deferring this until the library is wired into an actual sending/generation flow — at that point we'll cache parsed templates (cheap, just a `ConcurrentDictionary<(string, string), IFluidTemplate>`) and consider `IFileProvider.Watch(...)` for source invalidation. Not worth the invalidation complexity while there are zero callers.

File: apps/api/src/Eventuras.Libs.DocComposer/ServiceCollectionExtensions.cs [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+        this IServiceCollection services,
+        Action<DocComposerOptions> configure)
+    {
+        ArgumentNullException.ThrowIfNull(services);
+        ArgumentNullException.ThrowIfNull(configure);
+
+        services.AddOptions<DocComposerOptions>().Configure(configure);
+        services.AddSingleton<IDocumentComposer, FluidDocumentComposer>();
+        services.AddSingleton<IEmailComposer, FluidEmailComposer>();
+        return services;
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.DocComposer/ServiceCollectionExtensions.cs]: Fixed in a5fd6f7: `AddFluidDocComposer` now adds `.Validate(...)` rules for `TemplateFileProvider` and `DefaultLocale` plus `.ValidateOnStart()`, so misconfiguration fails fast at host startup.

File: apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+    private static FluidDocumentComposer CreateComposer(string defaultLocale = "en")
+    {
+        var templatesDir = Path.Combine(AppContext.BaseDirectory, "Templates");
+        var provider = new PhysicalFileProvider(templatesDir);
+        var options = Options.Create(new DocComposerOptions
+        {
+            TemplateFileProvider = provider,
+            DefaultLocale = defaultLocale
+        });
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs:26]: Fixed in a5fd6f7: test class now implements `IDisposable` and disposes the `PhysicalFileProvider` it creates. The constructor-validation test uses a local `using var` for the same reason.

File: apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidEmailComposerTest.cs [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+    private static FluidEmailComposer CreateComposer()
+    {
+        var templatesDir = Path.Combine(AppContext.BaseDirectory, "Templates");
+        var provider = new PhysicalFileProvider(templatesDir);
+        var options = Options.Create(new DocComposerOptions
+        {
+            TemplateFileProvider = provider,
+            DefaultLocale = "en"
+        });
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidEmailComposerTest.cs:26]: Fixed in a5fd6f7: test class now implements `IDisposable` and disposes the `PhysicalFileProvider`.

File: apps/api/src/Eventuras.Libs.DocComposer/PlainTextExtractor.cs [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.DocComposer/PlainTextExtractor.cs:175]: Declining — same as the prior occurrence of this rule. The existing `foreach (var raw in lines) { var line = CollapseInlineWhitespace(raw).TrimEnd(); ... }` is fine; pushing the projection into the loop header doesn't improve readability.

File: apps/api/src/Eventuras.Libs.DocComposer/PlainTextExtractor.cs [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/src/Eventuras.Libs.DocComposer/PlainTextExtractor.cs:175]: Declining — duplicate of the prior occurrence on this line. Foreach with an intermediate `var line = ...` is clearer than `.Select(...)` in the loop header here.

File: apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+
+        var result = await composer.ComposeAsync("welcome", model, "EN");
+
+        Assert.Equal("Welcome, Leo Losen!", result.Subject);
+    }
+
+    [Fact]
+    public void Constructor_Throws_WhenDefaultLocaleIsWhitespace()
+    {
+        var templatesDir = Path.Combine(AppContext.BaseDirectory, "Templates");
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs:94]: Declining — `"Templates"` is a hardcoded literal, not user input, so `Path.Combine` cannot silently drop the base directory here. False positive.

File: apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+namespace Eventuras.Libs.DocComposer.Tests;
+
+public class FluidDocumentComposerTest
+{
+    public sealed record WelcomeModel(string Name, string EventTitle, string Url);
+
+    private static FluidDocumentComposer CreateComposer(string defaultLocale = "en")
+    {
+        var templatesDir = Path.Combine(AppContext.BaseDirectory, "Templates");
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs:20]: Declining — `"Templates"` is a hardcoded literal, not user input. False positive.

File: apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+
+        var result = await composer.ComposeAsync("welcome", model, "EN");
+
+        Assert.Contains("Hello Leo Losen", result.Html);
+    }
+
+    [Fact]
+    public void Constructor_Throws_WhenDefaultLocaleIsWhitespace()
+    {
+        var templatesDir = Path.Combine(AppContext.BaseDirectory, "Templates");
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidDocumentComposerTest.cs:94]: Declining — `"Templates"` is a hardcoded literal, not user input. False positive.

File: apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidEmailComposerTest.cs [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace Eventuras.Libs.DocComposer.Tests;
+
+public class FluidEmailComposerTest
+{
+    public sealed record WelcomeModel(string Name, string EventTitle, string Url);
+
+    private static FluidEmailComposer CreateComposer()
+    {
+        var templatesDir = Path.Combine(AppContext.BaseDirectory, "Templates");
```
[the remark this answers is not in the document]
Comment (losolio) [reply apps/api/tests/Eventuras.Libs.DocComposer.Tests/FluidEmailComposerTest.cs:20]: Declining — `"Templates"` is a hardcoded literal, not user input. False positive.
